extensions/amp-story/1.0/amp-story-grid-layer.js

```diff
@@ -83,84 +66,26 @@ export class AmpStoryGridLayer extends AmpStoryBaseLayer {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
-
-    /** @private {?{horiz: number, vert: number}} */
-    this.aspectRatio_ = null;
-
-    /** @private {number} */
-    this.scalingFactor_ = 1;
   }
 
   /** @override */
   buildCallback() {
     super.buildCallback();
-    this.applyResponsivenessPresets_();
+    this.applyAspectRatioAttributes_();
     this.setOwnCssGridStyles_();
     this.setDescendentCssGridStyles_();
-    this.initializeListeners_();
   }
 
   /**
-   * Applies the attributes to the layer from the preset specified in the [preset] attribute.
+   * Grab the aspect-ratio attribute and apply to CSS variable as a fraction.
    * @private
    */
-  applyResponsivenessPresets_() {
-    if (!this.element.hasAttribute(PRESET_ATTRIBUTE_NAME)) {
-      return;
-    }
-    const preset = this.element.getAttribute(PRESET_ATTRIBUTE_NAME);
-    const presetDetails = GRID_LAYER_PRESET_DETAILS[preset];
-    if (!presetDetails) {
-      return;
-    }
-    Object.entries(presetDetails).forEach((keyValue) =>
-      this.element.setAttribute(keyValue[0], keyValue[1])
-    );
-  }
-
-  /** @private */
-  initializeListeners_() {
-    const aspectRatio = this.element.getAttribute('aspect-ratio');
-    const scalingFactorFloat = parseFloat(
-      this.element.getAttribute('scaling-factor')
-    );
-    if (scalingFactorFloat && scalingFactorFloat > 0) {
-      this.scalingFactor_ = scalingFactorFloat;
-    }
-    if (aspectRatio) {
-      const aspectRatioSplits = aspectRatio.split(':');
-      const horiz = parseInt(aspectRatioSplits[0], 10);
-      const vert = parseInt(aspectRatioSplits[1], 10);
-      if (horiz > 0 && vert > 0) {
-        this.aspectRatio_ = {horiz, vert};
-        const storeService = getStoreService(this.win);
-        storeService.subscribe(
-          StateProperty.PAGE_SIZE,
-          this.updatePageSize_.bind(this),
-          true /* callToInitialize */
-        );
-      }
-    }
-  }
-
-  /**
-   * @param {?{width: number, height: number}} pageSize
-   * @private
-   */
-  updatePageSize_(pageSize) {
-    if (!pageSize) {
-      return;
-    }
-    const {height: vh, width: vw} = pageSize;
-    const {horiz, vert} = this.aspectRatio_;
-    const width = Math.min(vw, (vh * horiz) / vert);
-    const height = Math.min(vh, (vw * vert) / horiz);
-    if (width > 0 && height > 0) {
-      this.getVsync().mutate(() => {
-        setStyles(this.element, {
-          '--i-amphtml-story-layer-width': px(width * this.scalingFactor_),
-          '--i-amphtml-story-layer-height': px(height * this.scalingFactor_),
-        });
+  applyAspectRatioAttributes_() {
+    if (this.element.hasAttribute('aspect-ratio')) {
+      setImportantStyles(this.element, {
+        '--aspect-ratio': this.element
```

Comment:
Thinking out loud: this might cause CLS but it'll be fine since we'll keep the service-renderer till we ship the optimizers/transformers that will create this CSS variable server-side.

Reply:
Yes you're right, if the CSS is added, this is one of the transformations we'd want to do in the AMP cache or AMP transformer. I'm keeping track of items like this that need to be in the transformers, and there's more context in the description of this PR for reference